lib/elixir/lib/kernel.ex

```diff
@@ -2924,6 +2926,10 @@ defmodule Kernel do
     end
   end
 
+  defp comp(left, {:|, _, [h, t]}) do
+    quote(do: :erlang.or(unquote(left) in unquote(t), :erlang."=:="(unquote(left), unquote(h))))
```

Comment:
For other cases order not that important but for this specific one I think we should put less costly check at first position.